src/Drips.sol

```diff
@@ -357,23 +357,26 @@ library Drips {
     /// @param defaults The list of default ends, must be sorted by start
     /// @param length The length of the list
     /// @param balance The balance available for drips without duration
-    /// @return end The end time of drips without duration
+    /// @return end_ The end time of drips without duration
     function _receiversDefaultEnd(
         DefaultEnd[] memory defaults,
-        uint256 length,
+        uint8 length,
         uint128 balance
-    ) private pure returns (uint32 end) {
+    ) private pure returns (uint32 end_) {
         uint32 lastStart = 0;
-        uint128 amtPerSec = 0;
-        end = type(uint32).max;
-        for (uint256 i = 0; i < length; i++) {
-            uint32 start = defaults[i].start;
+        uint136 amtPerSec = 0;
+        uint136 end = type(uint136).max;
+        for (uint8 i = 0; i < length; i++) {
+            DefaultEnd memory defaultEnd = defaults[i];
+            uint32 start = defaultEnd.start;
             if (start >= end) break;
-            balance -= amtPerSec * (start - lastStart);
+            balance -= uint128(amtPerSec * (start - lastStart));
```

Comment:
Do you think it makes sense to helper functions here?
```
safeMul(uint128, uint32)
```
Inside we would cast it to a `uint136` make the multiplication and cast it back?

Reply:
The point of this PR is to make this exact `amtPerSec` variable a `uint136`, because it needs to be able to handle a sum of all receivers' `amtPerSec`s. We'd need `safeMul(uint136, uint32) returns (uint128)`, which is quite confusing without the context and not very useful. We know that this casting is safe only because if `amtPerSec > balance`, `start == lastStart`, so the result is capped at `balance`, which is `uint128`.